infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -182,55 +181,54 @@ public TopologyConfigurationMsg queryTopologyConfig(boolean useCached) {
     }
 
     /**
-     * Change one of the standby as the primary and primary become the standby
+     * Change one of the standby as the primary and primary becomes the standby
      **/
-    public static TopologyDescriptor changePrimary(TopologyConfigurationMsg topologyConfig) {
-        TopologyDescriptor siteConfig = new TopologyDescriptor(topologyConfig);
-        ClusterDescriptor oldPrimary = new ClusterDescriptor(siteConfig.getActiveCluster(),
-                ClusterRole.STANDBY);
-        Map<String, ClusterDescriptor> standbys = new HashMap<>();
+    public static TopologyDescriptor changeActiveCluster(TopologyConfigurationMsg topologyConfig) {
+        TopologyDescriptor topologyDescriptor = new TopologyDescriptor(topologyConfig);
+
+        // Convert the current active to standby
+        ClusterDescriptor oldActive = topologyDescriptor.getActiveClusters().values().iterator().next();
+        ClusterDescriptor newStandby = new ClusterDescriptor(oldActive, ClusterRole.STANDBY);
+
+        List<ClusterDescriptor> standbyClusters = Arrays.asList(newStandby);
         ClusterDescriptor newPrimary = null;
-        ClusterDescriptor standby;
 
-        standbys.put(oldPrimary.getClusterId(), oldPrimary);
-        for (String endpoint : siteConfig.getStandbyClusters().keySet()) {
-            ClusterDescriptor info = siteConfig.getStandbyClusters().get(endpoint);
+        for (ClusterDescriptor standbyCluster : topologyDescriptor.getStandbyClusters().values()) {
             if (newPrimary == null) {
-                newPrimary = new ClusterDescriptor(info, ClusterRole.ACTIVE);
+                newPrimary = new ClusterDescriptor(standbyCluster, ClusterRole.ACTIVE);
             } else {
-                standby = new ClusterDescriptor(info, ClusterRole.STANDBY);
-                standbys.put(standby.getClusterId(), standby);
+                standbyClusters.add(standbyCluster);
             }
         }
 
-        TopologyDescriptor newSiteConf = new TopologyDescriptor(1, newPrimary, standbys);
+        TopologyDescriptor newSiteConf = new TopologyDescriptor(1L, Arrays.asList(newPrimary), standbyClusters);
         return newSiteConf;
     }
 
     /**
      * Testing purpose to generate cluster role change.
      */
     public static class SiteManagerCallback implements Runnable {
-        public boolean siteFlip = false;
-        DefaultClusterManager siteManager;
+        public boolean clusterRoleChange = false;
+        DefaultClusterManager clusterManager;
 
         SiteManagerCallback(DefaultClusterManager siteManagerAdapter) {
-            this.siteManager = siteManagerAdapter;
+            this.clusterManager = siteManagerAdapter;
```

Comment:
this is confusing.. The way I understand is that clusterManager=siteManager and clusterManagerAdapter=siteManagerAdapter(the plugin).  So how can we assign the adapter to the the clusterManager itself?  I think the variable names need to be changed..

Also, should the adapter be renamed to clusterManagerAdapter?

Reply:
its just the naming, I'll rename to make it clear!